ocaml/libs/clock/date.ml

```diff
@@ -104,27 +97,26 @@ let weekday ~year ~mon ~day =
   let m = mon + (12 * a) - 2 in
   (day + y + (y / 4) - (y / 100) + (y / 400) + (31 * m / 12)) mod 7
 
-let to_rfc822 ((year, mon, day), ((h, min, s), _), tz) =
+let to_rfc822 {t; tz} =
+  let (year, mon, day), ((h, min, s), _) =
+    Ptime.to_date_time ?tz_offset_s:tz t
+  in
   let timezone = match print_tz tz with "Z" -> "GMT" | tz -> tz in
   let weekday = weekday ~year ~mon ~day in
   Printf.sprintf "%s, %d %s %d %02d:%02d:%02d %s" days.(weekday) day
     months.(mon - 1)
     year h min s timezone
 
-let to_ptime_t t =
-  match to_dt t |> Ptime.of_date_time with
-  | Some t ->
+let to_ptime = function
+  | {t; tz= None} as d ->
+      Printf.printf
```

Comment:
We shouldn't be printing to `stdout` from a library, can we use `Logs` or xapi-log instead?

These warnings would likely go nowhere in our Dom0 setup (I think stdout is probably redirected to /dev/null), and we did have issues with messages on stderr in the past (we got some end of file errors printed, but it wasn't obvious where they were coming from, and they only got printed when a buffer limit was reached because channel output in OCaml is not line-buffered.

In particular this one lacks a terminating newline and a flush so we wouldn't see this message, unless the program gets shut down.

Reply:
Changed it to a warning in the logs